src/nft/components/nftExploreBanner/NftExploreBanner.tsx

```diff
@@ -0,0 +1,129 @@
+import { Trans } from '@lingui/macro'
+import { Box } from 'nft/components/Box'
+import { subhead } from 'nft/css/common.css'
+import { X } from 'react-feather'
+import { useNavigate } from 'react-router-dom'
+import { useShowNftPromoBanner } from 'state/user/hooks'
+import styled, { useTheme } from 'styled-components/macro'
+import { StyledInternalLink } from 'theme'
+import { Z_INDEX } from 'theme/zIndex'
+
+import nftPromoImage1 from '../nftExploreBanner/nftArt1.png'
+import nftPromoImage2 from '../nftExploreBanner/nftArt2.png'
+import nftPromoImage3 from '../nftExploreBanner/nftArt3.png'
+
+function getRandom(list: any[]) {
+  return list[Math.floor(Math.random() * list.length)]
+}
+const randomizedNftImage = getRandom([nftPromoImage1, nftPromoImage2, nftPromoImage3])
+
+const PopupContainer = styled.div<{ show: boolean }>`
+  background-color: ${({ theme }) => theme.backgroundSurface};
+  box-shadow: ${({ theme }) => theme.deepShadow};
+  border: 1px solid ${({ theme }) => theme.backgroundOutline};
+  border-radius: 12px;
+  cursor: pointer;
+  color: ${({ theme }) => theme.textPrimary};
+  height: 100px;
+  display: ${({ show }) => (show ? 'flex' : 'none')};
+  flex-direction: column;
+  bottom: 48px;
+  position: fixed;
+  right: clamp(0px, 1vw, 16px);
+  text-decoration: none;
+  width: ${Math.min(391, window.innerWidth - 5)}px;
```

Comment:
I think we could improve this a bit. Since we have CSS units such as vw, we should be able to refer to innerWidth of window without JavaScript. Also, we have calc and other functions to achieve what we want.

This will also get rid of the issue that @cbachmeier describes, because `window` is resolved once at the time of evaluating the CSS and is not updated dynamically. A good real life use case is changing rotation on a mobile.

Reply:
Also, I think it would be cleaner to use `@media` queries here.

For smaller devices, make it `width: 100vw` (or similar, depends on your padding/margin), otherwise, `391px`. Also, please make sure to follow `mobile-first` approach when writing media queries. This means the base CSS should define layout for smallest breakpoint and media queries should be starting with `min-width`.